### PR TITLE
[Anomaly Task] ✨ Upgrade anomalib

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
# Description

- Upgrade anomalib version

BMM- https://ci.iotg.sclab.intel.com/job/IMPT/job/IMPTOps/38559/

# 🧪
5 failed tests. Rest pass http://validationreports.sclab.intel.com:8006/reports/build_number_report?test_session_build_number=pr-ote-test-573&environment=iotg-dev-workstation-17
image

### CODE REVIEW

**Review by djdameln — DISMISSED** [review]

**Comment by ashwinvaidya17** [comment]
run ote-tests

**Comment by ashwinvaidya17** [comment]
run ote-test

File: external/anomaly/tasks/inference.py [hunk 1]
```diff
@@ -294,8 +294,9 @@ def save_model(self, output_model: ModelEntity) -> None:
     def _set_metadata(self, output_model: ModelEntity):
         output_model.set_data("image_threshold", self.model.image_threshold.value.cpu().numpy().tobytes())
         output_model.set_data("pixel_threshold", self.model.pixel_threshold.value.cpu().numpy().tobytes())
-        output_model.set_data("min", self.model.min_max.min.cpu().numpy().tobytes())
-        output_model.set_data("max", self.model.min_max.max.cpu().numpy().tobytes())
+        if hasattr(self.model, "normalization_metrics") and self.model.normalization_metrics.state_dict() is not None:
+            for key, value in self.model.normalization_metrics.state_dict().items():
```
Comment (djdameln) [inline external/anomaly/tasks/inference.py]: I guess this would work, but the openvino task assumes that the output model contains the "min" and "max" keys specifically. So this would not allow generalizing to other normalization methods.

Comment (ashwinvaidya17) [reply external/anomaly/tasks/inference.py]: Ah right! I think I'll revert it just for consistency.

Comment (djdameln) [reply external/anomaly/tasks/inference.py]: I think that would be safer

**Comment by ashwinvaidya17** [comment]
run ote-test

**Review by djdameln — APPROVED** [review]
I'm not sure if this new normalization design is the best approach from the OTE perspective, but we could discuss this later. Let's merge this for now to restore compatibility with the latest anomalib.

**Comment by goodsong81** [comment]
@ashwinvaidya17 @samet-akcay Could you confirm that this PR should be included in this release?

**Review by samet-akcay — APPROVED** [review]
thanks, would be good to pull these changes to the feature/otx branch
